checklist.html

```diff
@@ -1,242 +1,1029 @@
+      <summary>
+        Make sure that form input errors are displayed in list above the <code>form</code> after submission.
+      </summary>
+      <div class="checklist-details">
+        <p class="checklist-criterion">
+          <a href="https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-identified.html">3.3.1 Error Identification</a>
+        </p>
+        <p class="checklist-description">
+          This provides a way for assistive technology users to quickly have a high-level understanding of what issues are present in the form. This is especially important for larger forms with many inputs. Make sure that each reported error also has a link to the corresponding field with invalid input.
+        </p>
+      </div>
+    </details>
+    <details id="section-forms-05">
+      <summary>
+        Associate input error messaging with the input it corresponds to.
+      </summary>
+      <div class="checklist-details">
+        <p class="checklist-criterion">
+          <a href="https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-identified.html">3.3.1 Error Identification</a>
+        </p>
+        <p class="checklist-description">
+          Techniques such as <a href="https://developer.paciellogroup.com/blog/2018/09/describing-aria-describedby/">using <code>aria-describedby</code></a> allow people who use assistive technology to more easily understand the difference between the input and the error message associated with it.
+        </p>
+      </div>
+    </details>
+    <details id="section-forms-06">
+      <summary>
+        Make sure that error, warning, and success states are not visually communicated by just color.
+      </summary>
+      <div class="checklist-details">
+        <p class="checklist-criterion">
+          <a href="https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html">1.4.1 Use of Color</a>
+        </p>
+        <p class="checklist-description">
+          People who are color blind, who have other low vision conditions, or different cultural understandings for color may not see the state change, or understand what kind of feedback the state represents if color is the only indicator.
+        </p>
+      </div>
+    </details>
+    <details id="section-forms-07">
+      <summary>
+        Update the page or view's <code>title</code> to include high-level form feedback.
+      </summary>
+      <div class="checklist-details">
+        <p class="checklist-criterion">
+          <a href="https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html">1.3.1 Info and Relationships</a>
+        </p>
+        <p class="checklist-description">
+          The <code>title</code> element's content is often the first content announced by a screen reader. Adding feedback about the presence of things like errors or warnings after form submission allows people to know there are items on the page or view that need their attention.
```

Comment:
Thought I had mentioned something about this before, but updating the `title` element will not be announced to screen readers.  this sort of advice is only applicable if a form submission completely refreshes the page (non-spa) and the user has to renavigate the document as if they've gone to a new page.

Reply:
This assumed full page refresh. It's still good practice, but folks will need to take other action for SPAs to convey the same thing.

Reply:
We should explicitly call that out then since so many forms even outside of full SPAs could use JS to just update the current page, where this advice still wouldn't hold true without that clarity.

Reply:
I still think it's good UX to do, especially to make developers aware that it's A) something they can do and B) something that's useful. What's a better way to phrase this?

Reply:
>The text of the <code>title</code> element is typically one of the first announcements made by a screen reader when a new page loads. Adding concise feedback about the presence of errors or warnings, after a form submission has taken place, can help alert people that there are form fields that need attention.  Note that dynamically updating the text of a <code>title</code> element will not be announced to screen readers. SPAs will need to employ other methods to ensure such updates are announced by screen readers.